### PR TITLE
Add Javascript-free darkmode

### PR DESCRIPTION
**Pull request by out-of-phaze**
## Description of changes
Adds darkmode to both the chat and skin, using only winset and CSS.
Adds a pref to toggle it. Toggling it clears your chat so that the changes take effect instantly. (Alternatively, I could make it so that it'd only take effect on the next login, but that's kind of annoying.)

## Why and what will this PR improve
it's a good feature to have so that my eyes don't melt when I'm testing something at 3am and the light of God hits my retinas at full force

## Authorship
Me for code; Baystation12 for certain CSS color values and the basic darkmode code.

## Changelog
:cl:
add: Added dark mode to the UI, which can be toggled in global preferences and is on by default.
/:cl:

### CODE REVIEW

File: code/modules/client/darkmode.css [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+.selenian				{color: #22228b;}
+
+.interface				{color: #750e75;}
+
+.good					{color: #4f7529; font-weight: bold;}
+.bad					{color: #ee0000; font-weight: bold;}
+
+BIG IMG.icon 			{width: 32px; height: 32px;}
+
+/* Codex */
```
Comment (MistakeNot4892) [inline code/modules/client/darkmode.css]: Shouldn't need this stuff unless common.css isn't passed to browsers when darkmode is set.

File: code/modules/client/darkmode.dm [hunk 2]
```diff
@@ -0,0 +1,132 @@
+//Darkmode preference by Kmc2000//
+
+/*
+This lets you switch chat themes by using winset and CSS loading, you must relog to see this change (or rebuild your browseroutput datum)
+Things to note:
+If you change ANYTHING in interface/skin.dmf you need to change it here:
```
Comment (MistakeNot4892) [inline code/modules/client/darkmode.dm:9]: Wonder if we can unit test this somehow.

File: code/modules/client/lightmode.css [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+.selenian				{color: #22228b;}
+
+.interface				{color: #330033;}
+
+.good					{color: #00ff00; font-weight: bold;}
+.bad					{color: #ee0000; font-weight: bold;}
+
+BIG IMG.icon 			{width: 32px; height: 32px;}
+
+/* Codex */
```
Comment (MistakeNot4892) [inline code/modules/client/lightmode.css]: As above, looks like you don't need the codex classes.

File: code/modules/client/darkmode.dm [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+Format:
+winset(src, "window as appears in skin.dmf after elem", "var to change = currentvalue;var to change = desired value")
+*/
+
+/client/proc/deactivate_darkmode(clear_chat = TRUE) //There's no way round it. We're essentially changing the skin by hand. It's painful but it works, and is the way Lummox suggested.
+	//Main windows
+	winset(src, "infowindow", "background-color = [COLOR_DARKMODE_DARKBACKGROUND];background-color = none")
```
Comment (MistakeNot4892) [inline code/modules/client/darkmode.dm:16]: Not going to request this for your PR, but I feel like we could convert a lot of this into a set of static lists categorized by winset field and arguments.

File: code/modules/client/lightmode.css (hunk 3 above)
Comment (out-of-phaze) [reply code/modules/client/lightmode.css]: This was just copied from client/script, so yeah.

**Comment by MistakeNot4892** [comment]
Per convo on Discord, there's a couple of unstyled strings sent in client/New prior to CSS being sent.

**Review by MistakeNot4892 — APPROVED** [review]

**Pull request merged**